### PR TITLE
6375 Add mobile main site nav

### PR DESCRIPTION
**Pull request by TylerMatteo**
### Summary
This PR builds upon the existing Header and NavLink work to implement the mobile version of the main site nav. It basically "hides" the nav you see in the top right corner on mobile, in favor of showing the new "burger menu" drawer nav

#### Tasks/Bug Numbers
 - Fixes [AB#6375](https://dev.azure.com/NYCPlanning/cc280b0d-40a0-4689-b852-2e6247f1af50/_workitems/edit/6375)

### CODE REVIEW

File: src/components/Header/Header.tsx [hunk 1]
```diff
@@ -1,34 +1,112 @@
...
+            outline: "none",
+          }}
+          _focus={{
+            color: "teal.600",
+            outline: "none",
+          }}
+          mr={6}
+          transform={`scale(${isOpen ? "1.125, 1.0833" : "1, 1"})`}
+        />
+        <Drawer isOpen={isOpen} onClose={onClose} placement="left" size="xs">
```
Comment (godfreyyeung) [inline src/components/Header/Header.tsx:56]: It's a bit awkward that the Drawer is not hidden or shown based on breakpoint, but only isOpen. I see that the Hamburger IconButton is hidden based on breakpoint, and I get that the theory is Drawer is open/closed based on user interaction with that IconButton or  the DrawerOverlay. But it is leading to awkward behavior when a user opens the burger menu, but then resizes to a larger screen:
image


My first intuition  was that the Drawer should be hidden  based on breakpoint.

**Comment by godfreyyeung** [comment]
Not sure if it's just me but there's some strange highlighting going on when I first open the burger menu.
image

File: src/components/Header/Header.tsx (hunk 1 above)
Comment (TylerMatteo) [reply src/components/Header/Header.tsx:56]: Oh good catch, I didn't really test for someone going between sizes. I should be able to just hide in on desktop sizes.

**Comment by godfreyyeung** [comment]
> Not sure if it's just me but there's some strange highlighting going on when I first open the burger menu. image

Pretty sure this is a Chrome thing and we need to figure out how to use `-webkit` css to override this behavior

**Comment by TylerMatteo** [comment]
This is ready for another look. I added a bit of logic to listen for screen resizing and close the drawer on smaller screens. I fixed the blue outline thing by removing the default outline and adding a `focus` style to the buttons when they have focus so there's still a visual queue for keyboard nav users. Focused nav links get the green underline, but not the bigger size and grey bg that the active link gets. The first nav link is automatically focused when a user opens the drawer, that is intentional.
image

**Review by godfreyyeung — APPROVED** [review]

**Comment by godfreyyeung** [comment]
It's working for me now 👏🏾  Just one nit.... is it intentional that the breakpoint for the hamburger menu is different than the one for the drawer? 
image

**Review by godfreyyeung — APPROVED** [review]

**Comment by TylerMatteo** [comment]
> It's working for me now 👏🏾 Just one nit.... is it intentional that the breakpoint for the hamburger menu is different than the one for the drawer? image

No but I'm going to do another PR to get all this breakpoint stuff in sync.

**Pull request merged**